dist/pythonlibs/riotnode/riotnode/node.py

```diff
@@ -0,0 +1,205 @@
+        # I still want to show a PR before this
+        # I would prefer getting either no environment == os.environ or the
+        # full environment to use.
+        # It should also change the `TERM_STARTED_DELAY` thing.
+        self.env = os.environ.copy()
+        self.env.update(env or {})
+
+        self.term = None  # type: pexpect.spawn
+
+        self.logger = logging.getLogger(__name__)
+
+    @property
+    def application_directory(self):
+        """Absolute path to the current directory."""
+        return os.path.abspath(self._application_directory)
+
+    def board(self):
+        """Return board type."""
+        return self.env['BOARD']
+
+    def reset(self):
+        """Reset current node."""
+        # Ignoring 'reset' return value was taken from `testrunner`.
+        # For me it should not be done for all boards as it should be an error.
+        # I would rather fix it in the build system or have a per board
+        # configuration.
+
+        # Make reset yields error on some boards even if successful
+        # Ignore printed errors and returncode
+        self.make_run(self.RESET_TARGETS, stdout=DEVNULL, stderr=DEVNULL)
+
+    @contextlib.contextmanager
+    def run_term(self, reset=True, **startkwargs):
+        """Terminal context manager."""
+        try:
+            self.start_term(**startkwargs)
+            if reset:
+                self.reset()
+            yield self.term
+        finally:
+            self.stop_term()
+
+    def start_term(self, **spawnkwargs):
+        """Start the terminal.
+
+        The function is blocking until it is ready.
+        It waits some time until the terminal is ready and resets the node.
+        """
+        self.stop_term()
+
+        term_cmd = self.make_command(['term'])
+        self.term = self.TERM_SPAWN_CLASS(term_cmd[0], args=term_cmd[1:],
+                                          env=self.env, **spawnkwargs)
+
+        # on many platforms, the termprog needs a short while to be ready
+        time.sleep(self.TERM_STARTED_DELAY)
+
+    def _term_pid(self):
+        """Terminal pid or None."""
+        return getattr(self.term, 'pid', None)
+
+    def stop_term(self):
+        """Stop the terminal."""
+        with utils.ensure_all_subprocesses_stopped(self._term_pid(),
+                                                   self.logger):
+            self._safe_term_close()
+
+    def _safe_term_close(self):
+        """Safe 'term.close'.
+
+        Handles possible exceptions.
+        """
+        try:
+            self.term.close()
+        except AttributeError:
+            # Not initialized
+            pass
+        except ProcessLookupError:
+            self.logger.warning('Process already stopped')
+        except pexpect.ExceptionPexpect:
+            # Not sure how to cover this in a test
+            # 'make term' is not killed by 'term.close()'
+            self.logger.critical('Could not close make term')
```

Comment:
How about

```suggestion
            self.logger.critical('Could not close make term')
        finally:
            self.term = None
```

? Thes way one can check (e.g. in a subclass) if the terminal is running or not.

Reply:
I may even maybe put a `started` variable. As currently when starting we wait some time. I will see with the multi nodes handling.
Also, the `start` could just do nothing if it is already started.

However I did not really see a use case for calling `start` multiple times. In the current tests, the test function receive a `started` child. So I assumed to somehow have this initialization done before anything else.